syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+import torch
+
+from syft.frameworks.torch.overload_torch import overloaded
+from syft.frameworks.torch.tensors.interpreters import AbstractTensor
+
+
+class LargePrecisionTensor(AbstractTensor):
+    """LargePrecisionTensor allows handling of numbers bigger than LongTensor
+
+    Some systems using Syft require larger types than those supported natively. This tensor type supports arbitrarily
+    large values by packing them in smaller ones.
+    Typically a user will require to enlarge a float number by fixing its precision
+        tensor.fix_prec()
+
+    The large value is defined by `precision_fractional`.
+
+    The smaller values are of type `internal_type`. The split of the large number into the smaller values is done using
+    half the size of the selected internal type. This is to avoid overflow as types are in the range ±2**(size - 1).
```

Comment:
I don't think this is still true now that you use NumPy

Reply:
I think it still applies since the internal type is still a PyTorch type. NumPy is only used to perform the operations

Reply:
Ok but we don't have to worry about overflows anymore, do we?

Reply:
The overflow can happen when doing `divmod`. A way of seeing it happening is to remove the -1 in `def internal_precision(self)`

Reply:
I think the code is ok, I was just saying that for me, it's not half the size that is used but all of it, and that the `-1` is not here to avoid overflow but because we need to keep a bit for the sign in each chunk

Reply:
Haha... I see it now. You're right. Fixing!